### PR TITLE
V2 api

### PR DESCRIPTION
**Pull request by alicefuzier**
### What is this PR trying to achieve?
Have a v2 prefix in the api that serialises works using v2 display models. Depends on #1889 
Still TODO:
- ~Return correct error on wrong version( do we need to ?)~ For another PR
- [x] Add Swagger test to check that models returned are correct
- [x] Move ApiVersions object to a more suitable package
### Who is this change for?
🚨 💅

### CODE REVIEW

File: catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/controllers/WorksController.scala [hunk 1]
```diff
@@ -46,25 +45,27 @@ class WorksController @Inject()(
...
     } { request: MultipleResultsRequest =>
       val pageSize = request.pageSize.getOrElse(defaultPageSize)
       val includes = request.includes.getOrElse(WorksIncludes())
 
       for {
         resultList <- getWorkList(request, pageSize)
-        displayResultList = DisplayResultListV1(
+        displayResultList = DisplayResultList(
           resultList = resultList,
+          toDisplayWork,
```
Comment (alexwlchan) [inline catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/controllers/WorksController.scala]: nit: keyword arg pls?

File: catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/controllers/WorksController.scala [hunk 2]
```diff
@@ -237,13 +237,13 @@ class WorksController @Inject()(
     // in the public docs.
   }
 
-  private def setUpSingleWorkSwaggerDocs(version: String, doc: Operation) = {
+  private def setUpSingleWorkSwaggerDocs[T <: DisplayWork](version: ApiVersions.Value, doc: Operation)(implicit evidence: TypeTag[T])  = {
```
Comment (alexwlchan) [inline catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/controllers/WorksController.scala]: Making these be type parameters and commonising it all is super nice. 👌

File: catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/models/DisplayResultList.scala [hunk 3]
```diff
@@ -1,30 +1,29 @@
...
-import uk.ac.wellcome.platform.api.models.ResultList
+import uk.ac.wellcome.display.models.{DisplayWork, WorksIncludes}
+import uk.ac.wellcome.models.IdentifiedWork
 
 @ApiModel(
   value = "ResultList",
   description = "A list of things."
 )
-case class DisplayResultListV1(
+case class DisplayResultList[T <: DisplayWork](
```
Comment (alexwlchan) [inline catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/models/DisplayResultList.scala]: I didn’t know you could have type parameters on case classes – neat.

File: catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/ApiSwaggerTest.scala [hunk 4]
```diff
@@ -2,23 +2,44 @@ package uk.ac.wellcome.platform.api
...
+      val tree = readTree(s"/test/${version.toString}/swagger.json")
+      tree.at("/paths").isObject shouldBe true
+      tree.at("/paths").fieldNames.toList should contain only ("/works", "/works/{id}")
+    }
   }
 
-  it("should include the DisplayError model") {
-    val tree = readTree("/test/v1/swagger.json")
-    tree.at("/definitions/Error/type").toString should be("\"object\"")
+  it("should show the correct Work model for v1") {
```
Comment (alexwlchan) [inline catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/ApiSwaggerTest.scala]: and v2?

File: catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/ApiSwaggerTest.scala [hunk 5]
```diff
@@ -2,23 +2,62 @@ package uk.ac.wellcome.platform.api
...
-class ApiSwaggerTest extends FunSpec with Matchers with fixtures.Server {
+class ApiSwaggerTest
+    extends FunSpec
+    with Matchers
+    with fixtures.Server
+    with PropertyChecks {
 
-  it("should return a valid JSON response") {
-    val tree = readTree("/test/v1/swagger.json")
+  it("should return a valid JSON response for all api versions") {
```
[the remark this answers is not in the document]
Comment (alicefuzier) [reply catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/ApiSwaggerTest.scala]: 😄 just fixed it

**Pull request merged**